src/ledger/LedgerStateSnapshot.h

```diff
@@ -158,12 +165,43 @@ class LedgerSnapshot : public NonMovableOrCopyable
         return mGetter->getAccount(header, tx, AccountID);
     }
     LedgerEntryWrapper load(LedgerKey const& key) const;
+};
+
+// A subclass of LedgerSnapshot that contains additional snapshots necessary for
+// transaction validation.
+class ExtendedLedgerSnapshot : public LedgerSnapshot
```

Comment:
Is there a reason to keep both LedgerSnapshot and ExtendedLedgerSnapshot? If it's just scope, it's fine to clean this up in the follow-up, but wanted to confirm.